controllers/complianceeventsapi/auth.go

```diff
@@ -0,0 +1,165 @@
+) (bool, error) {
+	allManagedClusterRules, err := getAllManagedClusterRules(ctx, config, token, client)
+	if err != nil {
+		return false, err
+	}
+
+	rules, ok := allManagedClusterRules[managedClusterName]
+	if !ok {
+		return false, nil
+	}
+
+	if slices.Contains(rules, "get") || slices.Contains(rules, "*") {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+func getAllManagedClusterRules(ctx context.Context, config *rest.Config,
+	token string, client client.Client,
+) (map[string][]string, error) {
+	mangedClusterList := &clusterv1.ManagedClusterList{}
+
+	err := client.List(ctx, mangedClusterList)
+	if err != nil {
+		return nil, err
+	}
+
+	managedClusterNames := make([]string, 0, len(mangedClusterList.Items))
+
+	for _, mc := range mangedClusterList.Items {
+		managedClusterNames = append(managedClusterNames, mc.Name)
+	}
+
+	return getManagedClusterRules(config,
+		token, managedClusterNames)
+}
+
+func getManagedClusterRules(config *rest.Config,
+	token string, managedClusterNames []string,
+) (map[string][]string, error) {
+	saName := getServiceAccountName(token)
+	if saName == "" {
+		return nil, fmt.Errorf("failed to parse token")
+	}
+	// Token include Bearer prefix
+	config.Impersonate = rest.ImpersonationConfig{
+		UserName: saName,
+	}
+
+	kclient, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		log.Error(err, "Failed to get clientSet")
+
+		return nil, err
+	}
+
+	managedClusterGVK := schema.GroupResource{
+		Group:    "cluster.open-cluster-management.io",
+		Resource: "managedclusters",
+	}
+
+	// key is managedClusterName and value is verbs.
+	// ex: {"managed1": ["get"], "managed2": ["list","create"], "managed3": []}
+	return rbac.GetResourceAccess(kclient, managedClusterGVK, managedClusterNames, "")
+}
+
+func getServiceAccountName(token string) string {
+	jwtString := strings.TrimPrefix(token, "Bearer ")
+
+	// generic map to store parsed token
+	var claims map[string]interface{}
+
+	// decode JWT token without verifying the signature
+	tokenString, err := jwt.ParseSigned(jwtString)
+	if err != nil {
+		return ""
+	}
+
+	err = tokenString.UnsafeClaimsWithoutVerification(&claims)
+	if err != nil {
+		return ""
+	}
+
+	re, ok := claims["sub"].(string)
+	if !ok {
+		return ""
+	}
+
+	return re
+}
+
+func getWriteAuth(ctx context.Context, config *rest.Config, token, managedClusterName string,
+) (bool, error) {
+	saName := getServiceAccountName(token)
+	if saName == "" {
+		return false, fmt.Errorf("failed to parse token")
+	}
+
+	ssar := &authorizationv1.SelfSubjectAccessReview{
+		Spec: authorizationv1.SelfSubjectAccessReviewSpec{
+			ResourceAttributes: &authorizationv1.ResourceAttributes{
+				Group:       policyv1.GroupVersion.Group,
+				Version:     policyv1.GroupVersion.Version,
+				Resource:    "policies",
+				Verb:        "patch",
+				Namespace:   managedClusterName,
+				Subresource: "status",
+			},
+		},
+	}
+
+	// Token include Bearer prefix
+	config.Impersonate = rest.ImpersonationConfig{
```

Comment:
Like my other comment, we just want to use the token directly instead of impersonating.